pkg/manager/member/pd_failover.go

```diff
@@ -77,7 +83,18 @@ func (pf *pdFailover) Failover(tc *v1alpha1.TidbCluster) error {
 	}
 
 	healthCount := 0
+	pdsts, err := pf.setLister.StatefulSets(ns).Get(controller.PDMemberName(tcName))
+	if err != nil {
+		return err
+	}
+	podNames := sets.String{}
+	for _, ordinal := range helper.GetPodOrdinals(*pdsts.Spec.Replicas, pdsts).List() {
+		podNames.Insert(util.GetPodName(tc, v1alpha1.PDMemberType, ordinal))
+	}
 	for podName, pdMember := range tc.Status.PD.Members {
+		if !podNames.Has(podName) {
```

Comment:
I think we should update `tryToMarkAPeerAsFailure` instead of here, why ignore the non-managed members in the cluster health check?

Reply:
It's ok for me. updated.